src/geo/data-providers/geojson/geojson-data-provider.js

```diff
@@ -15,33 +15,37 @@ var GeoJSONDataProvider = function (vectorLayerView, layerIndex) {
 // TODO: We can extract each "generator" to an individual file so that this file doesn't get too BIG
 GeoJSONDataProvider.prototype._dataGeneratorsForDataviews = {
   category: function (features, options) {
+    var filter = this._vectorLayerView.renderers[this._layerIndex].filter;
     var columnName = options.column;
     var numberOfCategories = 5;
-
-    // TODO: There's probably a more efficient way of doing this
-    var groups = _.groupBy(features, function (feature) { return feature.properties[columnName]; });
-    var groupCounts = _.map(Object.keys(groups), function (key) { return [key, groups[key].length]; });
-    var sortedGroups = _.sortBy(groupCounts, function (group) { return group[1]; }).reverse();
+    var sortedGroups = filter.getColumnValues(columnName);
```

Comment:
I realized that `getColumnValues` [accepts two parameters](https://github.com/CartoDB/d3.cartodb/pull/94/files#diff-d2ddee121ba21440f473010082e89e72R84), but here we're just passing one. Why?